File: 1.1/src/cz/mg/vulkan/PFNvkCmdBeginRenderPass.java
package cz.mg.vulkan;

public class PFNvkCmdBeginRenderPass extends VkFunctionPointer {
    public PFNvkCmdBeginRenderPass() {
    }

    protected PFNvkCmdBeginRenderPass(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdBeginRenderPass(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdBeginRenderPass(long value) {
        setValue(value);
    }

    public PFNvkCmdBeginRenderPass(VkInstance instance) {
        super(instance, new VkString("vkCmdBeginRenderPass"));
    }

    public void call(VkCommandBuffer commandBuffer, VkRenderPassBeginInfo pRenderPassBegin, int contents){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pRenderPassBegin != null ? pRenderPassBegin.getVkAddress() : VkPointer.NULL, contents);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, long pRenderPassBegin, int contents);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdBeginRenderPass.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdBeginRenderPass_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong pRenderPassBegin, jlong contents){
    (void)env;
    (void)jc;
    PFN_vkCmdBeginRenderPass f = (PFN_vkCmdBeginRenderPass)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        ((VkRenderPassBeginInfo*)jniLongToPointer(pRenderPassBegin)),
        *((VkSubpassContents*)jniLongToPointer(contents))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdBeginRenderPass_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong pRenderPassBegin, jint contents){
    (void)env;
    (void)jc;
    PFN_vkCmdBeginRenderPass f = (PFN_vkCmdBeginRenderPass)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        ((VkRenderPassBeginInfo*)jniLongToPointer(pRenderPassBegin)),
        (VkSubpassContents)contents
    );
}

